src/Templates/MethodTemplate.cshtml

```diff
@@ -161,33 +131,52 @@ func (client @(Model.Owner)) @(Model.PreparerMethodName)(@(Model.MethodParameter
     }
 </text>
 }
-
-    @(Model.IsLongRunningOperation()
-        ? "return preparer.Prepare(&http.Request{Cancel: cancel})"
-        : "return preparer.Prepare(&http.Request{})")
+    return preparer.Prepare(&http.Request{})
 }
 
+@{ 
+    var senderRetType = "*http.Response";
+    if (Model.IsLongRunningOperation())
+    {
+        senderRetType = Model.MethodReturnType;
+    }
+}
 
 @EmptyLine
 // @(Model.SenderMethodName) sends the @(Model.Name) request. The method will close the
 // http.Response Body if it receives an error.
-func (client @(Model.Owner)) @(Model.SenderMethodName)(req *http.Request) (*http.Response, error) {
+func (client @(Model.Owner)) @(Model.SenderMethodName)(req *http.Request) (@senderRetType, error) {
+@if (Model.IsLongRunningOperation())
+{
+<text>
+    sender := autorest.DecorateSender(client, azure.DoRetryWithRegistration(client.Client))
```

Comment:
`DoRetryWithRegistration` should not be used in all cases

Reply:
Good point, this should be predicated on Model.RegisterRP.

Reply:
Fixed.